src/simtools/simulator.py

```diff
@@ -644,26 +674,36 @@ def pack_for_register(self, directory_for_grid_upload=None):
         )
         directory_for_grid_upload.mkdir(parents=True, exist_ok=True)
 
-        tar_file_name = directory_for_grid_upload.joinpath(tar_file_name)
+        # If there are more than one model version,
+        # duplicate the corsika log file to have one for each model version with the "right name".
+        if len(self.array_models) > 1 and corsika_log_files:
```

Comment:
My feeling is that sonarq will definitely complain about the level of complexity of the `pack_for_register` function. Any change to simplify? Making it more modular?

Reply:
I actually separated the code in `_copy_corsika_log_file_for_all_versions` in particular because of that. The rest of the changes do not add much complexity, we just loop over the models. I could separate that to a different function as well, but I would prefer to wait for SQ to complain before I do it...